census/src/main/java/org/openjdk/skara/census/Namespace.java

```diff
@@ -91,4 +91,21 @@ static Namespace parse(Element ele, Map<String, Contributor> contributors) throw
 
         return new Namespace(name, mapping, reverse);
     }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
```

Comment:
The "same type" check could be phrased differently using instanceof, but what you have here should work okay too.

Reply:
Hm, I let the IDE generate these for me so didn't think much of it. Aren't there subtle differences with instanceof, if there are subtypes involved? Not that it matters in this particular instance.

Reply:
So I experimented some more. In the intellij wizard for this, if I check "accept subclasses", it generates code with `instanceof` instead. I don't think this equals method should do that.

Reply:
To circle back on this, there is a school of thought that getClass should not be used in this fashion in an equals method since it violates the Liskov substitution principle -- instances of a subclass can be used where instances of the superclass can used. "Effective Java, 3rd edition" covers this topic in item 10. However, in a case like this where (I assume) subclasses are not expected, it doesn't matter very much in practice.